sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java

```diff
@@ -0,0 +1,201 @@
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (isPrimitive(value)) {
+        simpleEntries.put(key, value);
+        continue;
+      }
+      if (isPrimitiveList(value)) {
+        simpleEntries.put(key, ((List<?>) value).stream().map(String::valueOf).collect(toList()));
+        continue;
+      }
+      if (isListOfMaps(value)) {
+        List<StructuredConfigProperties> list =
+            ((List<Map<String, Object>>) value)
+                .stream().map(YamlStructuredConfigProperties::new).collect(toList());
+        listEntries.put(key, list);
+        continue;
+      }
+      if (isMap(value)) {
+        YamlStructuredConfigProperties configProperties =
+            new YamlStructuredConfigProperties((Map<String, Object>) value);
+        mapEntries.put(key, configProperties);
+        continue;
+      }
+      throw new ConfigurationException(
+          "Unable to initialize ExtendedConfigProperties. Key \""
+              + key
+              + "\" has unrecognized object type "
+              + value.getClass().getName());
+    }
+  }
+
+  private static boolean isPrimitiveList(Object object) {
+    if (object instanceof List) {
+      List<?> list = (List<?>) object;
+      return list.stream().allMatch(YamlStructuredConfigProperties::isPrimitive);
+    }
+    return false;
+  }
+
+  private static boolean isPrimitive(Object object) {
+    return object instanceof String
+        || object instanceof Integer
+        || object instanceof Long
+        || object instanceof Float
+        || object instanceof Double
+        || object instanceof Boolean;
+  }
+
+  private static boolean isListOfMaps(Object object) {
+    if (object instanceof List) {
+      List<?> list = (List<?>) object;
+      return list.stream()
+          .allMatch(
+              entry ->
+                  entry instanceof Map
+                      && ((Map<?, ?>) entry)
+                          .keySet().stream().allMatch(key -> key instanceof String));
+    }
+    return false;
+  }
+
+  private static boolean isMap(Object object) {
+    if (object instanceof Map) {
+      Map<?, ?> map = (Map<?, ?>) object;
+      return map.keySet().stream().allMatch(entry -> entry instanceof String);
+    }
+    return false;
+  }
+
+  @Nullable
+  @Override
+  public String getString(String name) {
+    Object value = simpleEntries.get(name);
+    if (value instanceof String) {
+      return (String) value;
+    }
+    return null;
+  }
+
+  @Nullable
+  @Override
+  public Boolean getBoolean(String name) {
+    Object value = simpleEntries.get(name);
+    if (value instanceof Boolean) {
+      return (Boolean) value;
+    }
+    return null;
+  }
+
+  @Nullable
+  @Override
+  public Integer getInt(String name) {
+    Object value = simpleEntries.get(name);
+    if (value instanceof Integer) {
+      return (Integer) value;
+    }
+    if (value instanceof Long) {
+      return ((Long) value).intValue();
```

Comment:
🙈

Reply:
Probably worth a comment that calls out the loss of resolution if the value is actually a long.

Reply:
In a local version I got rid of the `getInt` versions of these methods for this reason. Figured that because the use case is config, its probably never going to be a problem to use the extra memory for the 64 bit long version.

I caved and restored the getInt accessors because sometimes you want to get a property like a port, and being forced to get a long and convert it down to an int to match the argument requirements of whatever library is being configured pushes the conversion problem to the caller.

I'll add a comment to the javadoc of getInt.